### PR TITLE
PostgreSQL version 10.11

### PR DESCRIPTION
**Pull request by DigitalBox98**
_Motivation:_  Propose PostgreSQL to versions 10.11

- In order to avoid any conflict with existing Postgresql 9 installed by default on Synology, port 5433 is used instead of 5432
- Default database in /var/packages/postgresql/target/var/data/ and config file is  /var/packages/postgresql/target/var/data/postgresql.conf
- PostgreSQL admin user "pgadmin" added by default in order to define a user password
- Wizard UI at installation time in order to define PostgreSQL admin user/password
- To check that version use : /var/packages/postgresql/target/bin/psql -p 5433 -U pgadmin -d postgres

### Checklist
- [ ] Build rule successfully for at least Braswell architecture (not checked yet other architectures)
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

**Comment by hgy59** [comment]
@DigitalBox98 please rebase your branch to current master (to exclude the redis related commits from this PR)

**Comment by hgy59** [comment]
@DigitalBox98 please remove the file `spksrc/postgresql/src/._install_uifile` and provide at least a wizard file for `eng` language.

File: cross/postgresql/Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+PKG_NAME = postgresql
+PKG_VERS = 10.11
+PKG_EXT = tar.gz
+PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = https://ftp.postgresql.org/pub/source/v10.11/
```
Comment (hgy59) [inline cross/postgresql/Makefile]: to simplify future updates use `v$(PKG_VERS)` instead of `v10.11` here

**Comment by hgy59** [comment]
This is a precondition for #3815 since postgresql 9.3 is not supported by TimescaleDB

File: cross/postgresql/Makefile (hunk 1 above)
Comment (DigitalBox98) [reply cross/postgresql/Makefile]: indeed fixed

**Comment by DigitalBox98** [comment]
> @DigitalBox98 please remove the file `spksrc/postgresql/src/._install_uifile` and provide at least a wizard file for `eng` language.

OK corresponding file removed. 
Regarding the "eng" language, the corresponding file is "install_uifile", so not sure to be the point here

**Comment by DigitalBox98** [comment]
I have some conflicts with rebasing. To make things simplier, I will fork again and apply a fresh new PR for PostgreSQL

**Comment by DigitalBox98** [comment]
You can close this PR and check the new ones with all fixes.

I didn't propose latest version (12 o 11) because they are not compiling without some manual patches. 
Version 10 is a good candidate because compilation is fine without additional effort, and is more up to date than current version 9 from Synology :)

**Comment by hgy59** [comment]
@DigitalBox98 I already prepared the review one hour ago but was verifying the cross compilation in between - and all-supported succeeded with my proposals.

**Comment by hgy59** [comment]
you'd better updated this PR instead of creating a new one ;-(

And this is the code to build postgresql 11.6:
add to cross/postgresql/Makefile:

```
# If you want to invoke the build from another makefile rather than manually, 
# you must unset MAKELEVEL or set it to zero.
# see: INSTALL file
ENV += MAKELEVEL=
```

**Comment by hgy59** [comment]
unfortunately the source branch is gone...

File: spk/postgresql/PLIST [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
[the remark this answers is not in the document]
Comment (DigitalBox98) [reply spk/postgresql/PLIST]: ok fixed on new PR

**Comment by DigitalBox98** [comment]
Sorry for the branch, I had some issues with the rebasing that's why I decided to start from scratch again. I will need to improve my knowledge on how to deal with branch & rebasing to avoid this in the future.

Thanks for all this great feedback, I've checked again with all the remarks and I confirm postgresql 10.11 is compiling fine and working as a charm on the NAS ;)

I will commit all these remarks to the new PR created. 
It's a good news to have a more recent version of postgresql available :)

**Comment by hgy59** [comment]
Yeah I am fighting with the `git` from time to time too.

Your new PR is based on your master branch. It is better to use a feature branch (like `postgresql`).
So you can keep your master in sync with the upstream master (synocommunity repository).

_Better never merge your branches into your master, as you will get out of sync with the upstream.
After I did so, I had to recreate (hard reset) my fork to get in sync again._

**Pull request closed without merging**